src/Shared/FileMatcher.cs

```diff
@@ -1673,7 +1673,7 @@ internal static bool IsMatch(string input, string pattern)
         /// <param name="pattern">Pattern against which string is matched.</param>
         internal static bool IsMatch(ReadOnlySpan<char> input, string pattern)
         {
-            if (input == null)
+            if (input == ReadOnlySpan<char>.Empty) // Comparing ReadOnlySpan with null in this scenario transforms the null to ReadOnlySpan Empty (https://github.com/dotnet/runtime/issues/84265)
             {
                 throw new ArgumentNullException(nameof(input));
```

Comment:
What's the right exception type to throw? Should this use `ThrowInternalError` instead?

Reply:
I reverted the change back to null detection, to resolve the failing tests first.  I'll investigate if we even need this check

Reply:
Internal error added.